targets/TARGET_STM/TARGET_STM32H7/flash_api.c

```diff
@@ -133,8 +142,15 @@ int32_t flash_program_page(flash_t *obj, uint32_t address, const uint8_t *data,
         }
     }
 
+#if defined(DUAL_CORE)
+#if defined(CORE_CM7)
     SCB_CleanInvalidateDCache_by_Addr((uint32_t *)StartAddress, FullSize);
```

Comment:
This part does not look right? We're anyway always doing the same thing?

Reply:
Either there should be "something else" in the `#else` part or these should not be any `#ifdef` magic here.

Reply:
SCB_CleanInvalidateDCache_by_Addr only exist for M7 core.
- for H743: DUAL_CORE is not defined
- for H747: DUAL_CORE is defined, then core side is checked